upath/core.py

```diff
@@ -342,6 +378,24 @@ def readlink(self):
     def touch(self, truncate=True, **kwargs):
         self._accessor.touch(self, truncate=truncate, **kwargs)
 
+    def mkdir(self, mode=0o777, parents=False, exist_ok=False):
+        """
+        Create a new directory at this given path.
+        """
+        if parents:
+            self._accessor.mkdir(
+                self,
+                create_parents=parents,
+                exist_ok=exist_ok,
+                permissions=mode,
+            )
+        else:
+            try:
+                self._accessor.mkdir(self, create_parents=False)
+            except OSError:
+                if not exist_ok or not self.is_dir():
```

Comment:
should not you check that for the exist_ok=True case it's actually an already exist error/exception instance ?

Reply:
I think you're correct. Changed :+1: